src/main/fc/rc_adjustments.c

```diff
@@ -459,7 +459,8 @@ void processRcAdjustments(controlRateConfig_t *controlRateConfig)
             newValue = applyStepAdjustment(controlRateConfig, adjustmentFunction, delta);
             pidInitConfig(pidProfile);
         } else if (adjustmentState->config->mode == ADJUSTMENT_MODE_SELECT) {
-            const uint16_t rangeWidth = ((2100 - 900) / adjustmentState->config->data.switchPositions);
+            const int switchPositions = systemConfig()->rateProfile6PosSwitch ? 6 : adjustmentState->config->data.switchPositions;
```

Comment:
This won't work in this way, since it will break this functionality for all other adjustment range options that use `switchPositions`. I like the idea of using a parameter to change the behaviour, but you will at least have to check and only apply this if it is the rate profile adjustment switch - but even then, doing a case by case this deep in the logic is not ideal.